src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/WhereCQLClauseAnalyzer.java

```diff
@@ -0,0 +1,80 @@
+package io.stargate.sgv2.jsonapi.service.operation.tables;
+
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.TableSchemaObject;
+import io.stargate.sgv2.jsonapi.service.operation.query.DBFilterBase;
+import io.stargate.sgv2.jsonapi.service.operation.query.DBLogicalExpression;
+import io.stargate.sgv2.jsonapi.service.operation.query.TableFilter;
+import io.stargate.sgv2.jsonapi.service.operation.query.TableFilterAnalyzedUsage;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The purpose for this analyzer class is to gather all the analyze result {@link
+ * TableFilterAnalyzedUsage} from all {@link TableFilter} in the TableWhereCQLClause. And then
+ * proceed with overall analyze.
+ *
+ * <p>E.G. {"filter": {"name": "Lisa", "age":25}}. <br>
+ * This API filter will resolve as two TableFilters against a table. And since name column is not
+ * indexed, the corresponding tableFilter will result a {@link TableFilterAnalyzedUsage} which
+ * contains information like "ALLOW FILTERING" and "warning msg". Then after
+ * tableWhereCQLClauseAnalyzer gathered all the {@link TableFilterAnalyzedUsage}, it will make final
+ * decision of analyzing the TableWhereCQLClause.
+ *
+ * <p>TODO, this class will be helpful for future features, such as TableFilter and Index
+ * recommendation.
+ */
+public class WhereCQLClauseAnalyzer {
```

Comment:
once merged, we will work on this refactor https://github.com/stargate/data-api/issues/1485